tests/emit.spec.ts

```diff
@@ -364,4 +364,35 @@ describe('emitted', () => {
     expect(wrapper.emitted().bar[0]).toEqual(['mounted'])
     expect(wrapper.emitted().bar[1]).toEqual(['click'])
   })
+
+  it('does not clear all emitted event history on mount/unmount', async () => {
+    const Foo = defineComponent({
+      name: 'Foo',
+      emits: ['foo'],
+      setup(_, ctx) {
+        return () =>
+          h(
+            'div',
+            {
+              onClick: () => {
+                ctx.emit('foo', 'bar')
+              }
+            },
+            'hello world'
+          )
+      }
+    })
+
+    const wrapper1 = mount(Foo)
+    await wrapper1.trigger('click')
+    expect(wrapper1.emitted('foo')).toHaveLength(1)
```

Comment:
Should we also assert 

```js
expect(wrapper2.emitted('foo')).toHaveLength(0)
```

Here as well?

Reply:
We would have to wait until after wrapper2 is mounted. We can do that, but the ultimate point of the test is to ensure that subsequent mounts no longer clear the emit history for previously mounted components.